pyhf/pdf.py

```diff
@@ -379,20 +383,9 @@ def expected_data(self, pars, include_auxdata=True):
         return tensorlib.concatenate(tocat)
 
     def constraint_logpdf(self, auxdata, pars):
-        tensorlib, _ = get_backend()
-        # iterate over all constraints order doesn't matter....
-        start_index = 0
-        summands = None
-        for cname in self.config.auxdata_order:
-            modifier, modslice = self.config.modifier(cname), \
-                self.config.par_slice(cname)
-            modalphas = modifier.alphas(pars[modslice])
-            end_index = start_index + int(modalphas.shape[0])
-            thisauxdata = auxdata[start_index:end_index]
-            start_index = end_index
-            constraint_term = tensorlib.log(modifier.pdf(thisauxdata, modalphas))
-            summands = constraint_term if summands is None else tensorlib.concatenate([summands,constraint_term])
-        return tensorlib.sum(summands) if summands is not None else 0
+        normal  = self.prepped_constraints_gaussian.logpdf(auxdata,pars)
+        poisson = self.prepped_constraints_poisson.logpdf(auxdata,pars)
+        return normal + poisson
```

Comment:
is it really just a sum? I thought it was multiplied together.

Reply:
it's logpdf, so sum is correct